src/main/webapp/app/exercises/programming/shared/instructions-render/extensions/programming-exercise-task.extension.ts

```diff
@@ -24,12 +26,7 @@ export class ProgrammingExerciseTaskExtensionWrapper implements ArtemisShowdownE
     // unique index, even if multiple tasks are shown from different problem statements on the same page (in different tabs)
     private taskIndex = 0;
 
-    constructor(
-        private programmingExerciseInstructionService: ProgrammingExerciseInstructionService,
-        private componentFactoryResolver: ComponentFactoryResolver,
-        private appRef: ApplicationRef,
-        private injector: Injector,
-    ) {}
+    constructor(private programmingExerciseInstructionService: ProgrammingExerciseInstructionService, private appRef: ApplicationRef, private injector: Injector) {}
```

Comment:
ApplicationRef is never used in this class (except in old & commented out code).
Please remove it if it is not needed anymore.